src/redux/superadmins/thunks.js

```diff
@@ -1,18 +1,92 @@
-import { getSuperadminsPending, getSuperadminsError, getSuperadminsSuccess } from './actions';
-
-export const getSuperAdmins = () => {
-  return async (dispatch) => {
-    try {
-      dispatch(getSuperadminsPending());
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/`);
-      const responseJson = await response.json();
-      const data = responseJson.data;
-      if (responseJson.error) {
-        throw new Error(responseJson.message);
+import * as actions from './actions';
+
+export const getSuperAdmins = async (dispatch) => {
+  dispatch(actions.getSuperadminsPending());
+  try {
+    const res = await fetch(`${process.env.REACT_APP_API_URL}/api/super-admins/`);
+    const data = await res.json();
+    dispatch(actions.resetInitialState());
+
+    if (res.status === 200) {
+      dispatch(actions.getSuperadminsSuccess(data.data));
```

Comment:
You can destructure data as { data, message }, instead of data.data or data.message